lib/optimize/SideEffectsFlagPlugin.js

```diff
@@ -117,5 +116,34 @@ class SideEffectsFlagPlugin {
 			});
 		});
 	}
+
+	static moduleHasSideEffects(modulePath, flagValue) {
+		switch(typeof flagValue) {
+			case "boolean":
+				return flagValue;
+			case "string":
+				{
+					if(flagValue === "false" || flagValue === "true") {
```

Comment:
It's kind of weird to support wrong booleans `"false"`. Remove this.

Reply:
Supporting `"false"` in  #6074 was the genesis of this PR.  Should we drop support for it?